apps/vscode/command_client/command_client.py

```diff
@@ -197,20 +198,30 @@ def run_command(
     return decoded_contents["returnValue"]
 
 
+command_server_directory = None
+
+
+# NB: See https://github.com/talonhub/community/issues/966 for why we do OS-specific temp dirs
+def get_platform_specific_communication_dir_path():
+    home_dir = Path(os.path.expanduser("~"))
+    directory_name = actions.user.command_server_directory()
+    if app.platform == "linux" or app.platform == "mac":
+        return home_dir / ".talon" / " .comms" / directory_name
+    elif app.platform == "windows":
+        # subprocess.run(["attrib","+H","myfile.txt"],check=True)
```

Comment:
This seems like it may have been some testing code?

Reply:
I left this in for someone with access to Windows in case they want to test using a hidden folder, but didn't uncomment it because I can't test if it works.

Reply:
@knausj85 any thoughts about that part? I'm happy to just delete the commented out bit if windows users don't care to add it

Reply:
I have no concerns with leaving this file unhidden. Thoughts on this @AndreasArvidsson ?

Reply:
fyi I'm not sure I want plugins making top-level directories under .talon, hidden or not

Reply:
@knausj85 I'm personally not a big fan of hidden files in windows so I'm fine with it being visible. With that being said I'm not sure placing it in the Talon directory is the right choice.

Reply:
Can those of you who have opinions try to decide on a path that is acceptable and I'll update the PRs with whatever it is? I don't care where it is, as long as those who do care agree on something.

Reply:
Sounds good. Should we continue that discussion in the issue or here?

Personally I would prefer to keep it in the temp directory and agree with @rntz arguments in the issue that the extension should respect `TMPDIR`.

If that doesn't work for some technical reason we should probably just create a new directory in user home. eg `~/.rpc`.

Reply:
I believe TMPDIR doesn't work as a solution because it can be different path in talon vs vscode. This can definitely be the case on nix, as noted here https://github.com/talonhub/community/issues/966#issuecomment-1236134227 and also unrelated to that plugin which I've run into (which I already forget where as I've been using this PR for ~ 1 year). I originally hardcoded /tmp in the PR, but then eventually we opted for a /home-based path, which unfortunately I now can't find the reason/comment for it seems.

But I think because of all the potential snap/flatpak/nix related issues mentioned it's likely just best to avoid hoping one of those tmpdir/runtime path solutions work everywhere? Pretty sure atm we don't have the capacity to actually test it.

Something like ~/.rpc is fine with me if that's what people like (I just want it fixed at this point :D), but imo it does seem a bit opaque and potentially collision prone vs some name that associates it with talon, which was why I chose to drop the folder in ~/.talon/... ~/.talon-rpc or similar maybe makes more sense?

Reply:
If someone on purpose has set a different environment variable in Talon and vscode they kind off asking for trouble. As long as we check the sources in the same order we should get the same result. 

Year that is a real problem because we don't want to break it for existing users.

`~/.talon-rpc` would be fine with me, but I'm not sure the home directory is any more stable than the temp directory since the home directory can also be moved with environment variables.

Reply:
FWIW I agree if it were actually the user just wantonly changing environment variables, it shouldn't something that needs catering to. However, this isn't a case of a user choosing to change TMPDIR, it's being done somewhere behind the scenes, so it's not something they can just choose not to do. 

In addition to the original bug filer, there were numerous (https://github.com/talonhub/community/issues/966#issuecomment-1536142000) cases of it differing in practice, as well as in my systems and for some other Nix users. When it started breaking for me (which prompted finally creating the PR) it's certainly not from me explicitly changing TMPDIR myself. This comment https://github.com/talonhub/community/issues/966#issuecomment-1293379354 also implies this type of thing happens on windows.

In light of that, I don't see how keeping the order checking consistent is enough, unless you mean you have some other mechanism to ensure the paths they are checking in the same order are actually matching? Maybe I'm missing something.